azurerm/resource_arm_sql_database.go

```diff
@@ -118,128 +119,157 @@ func resourceArmSqlDatabase() *schema.Resource {
-	}
+	createMode := sql.CreateMode(d.Get("create_mode").(string))
 
-	if v, ok := d.GetOk("edition"); ok {
-		command.Edition = azure.String(v.(string))
-	}
+	maxSizeBytes := d.Get("max_size_bytes").(string)
 
-	if v, ok := d.GetOk("collation"); ok {
-		command.Collation = azure.String(v.(string))
-	}
+	edition := sql.DatabaseEdition(d.Get("edition").(string))
 
-	if v, ok := d.GetOk("max_size_bytes"); ok {
-		command.MaxSizeBytes = azure.String(v.(string))
-	}
+	requestedServiceObjectiveIDString := d.Get("requested_service_objective_id").(string)
+	var requestedServiceObjectiveID uuid.UUID
 
-	if v, ok := d.GetOk("source_database_deletion_date"); ok {
-		command.SourceDatabaseDeletionDate = azure.String(v.(string))
-	}
+	requestedServiceObjectiveNameString := d.Get("requested_service_objective_name").(string)
+	var requestedServiceObjectiveName sql.ServiceObjectiveName
+
+	if requestedServiceObjectiveIDString != "" && requestedServiceObjectiveNameString == "" {
+
+		var errUUID error
+		requestedServiceObjectiveID, errUUID = uuid.FromString(requestedServiceObjectiveIDString)
+		if errUUID != nil {
+			return errUUID
+		}
+
+	} else if requestedServiceObjectiveIDString == "" && requestedServiceObjectiveNameString != "" {
+
+		requestedServiceObjectiveName = sql.ServiceObjectiveName(requestedServiceObjectiveNameString)
+
+	} else {
+
+		return fmt.Errorf("either service objective name or id must be specified")
 
-	if v, ok := d.GetOk("requested_service_objective_id"); ok {
-		command.RequestedServiceObjectiveID = azure.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("elastic_pool_name"); ok {
-		command.ElasticPoolName = azure.String(v.(string))
+	tags := d.Get("tags").(map[string]interface{})
+	metadata := expandTags(tags)
+
+	props := sql.DatabaseProperties{
+		Collation:  &collation,
+		Edition:    edition,
+		CreateMode: createMode,
 	}
 
-	if v, ok := d.GetOk("requested_service_objective_name"); ok {
-		command.RequestedServiceObjectiveName = azure.String(v.(string))
+	if createMode == sql.Default && maxSizeBytes != "" {
+		props.MaxSizeBytes = &maxSizeBytes
+		props.SourceDatabaseID = nil
+	} else if sourceDatabaseID != "" {
+		props.MaxSizeBytes = nil
+		props.SourceDatabaseID = &sourceDatabaseID
 	}
 
-	createRequest := rivieraClient.NewRequest()
-	createRequest.Command = command
+	if requestedServiceObjectiveName != "" {
+		props.RequestedServiceObjectiveName = requestedServiceObjectiveName
+	} else {
+		props.RequestedServiceObjectiveID = &requestedServiceObjectiveID
+	}
 
-	createResponse, err := createRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error creating SQL Database: %s", err)
+	if requestedServiceObjectiveNameString == "ElasticPool" {
+		props.ElasticPoolName = &elasticPoolName
 	}
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating SQL Database: %s", createResponse.Error)
+
+	parameters := sql.Database{
+		Name:               &databaseName,
```

Comment:
can we remove this, as it's specified in the URL generated from the CreateOrUpdate method below